src/server/JasmineGraphInstanceService.cpp

```diff
@@ -4181,36 +4182,46 @@ map<long, double> calculateLocalPageRank(string graphID, double alpha, string pa
-                                                                                    localGraphMap, workerSockets,
-                                                                                    worldOnlyVertexCount);
-
-    map<long, unordered_set<long>> hMapFromVerticesWorldToLocalFLow = getEdgesWorldToLocal(graphID, partitionID,
-                                                                                           serverPort, graphVertexCount,
-                                                                                           localDB, centralDB,
-                                                                                           localGraphMap,
-                                                                                           workerSockets);
-
     long entireGraphSize = atol(graphVertexCount.c_str());
     float mu = (damp / entireGraphSize);
     unordered_map<float, float> resultTreeMap;
     // calculating local pagerank
     map<long, double> rankMap;
 
-    // add world to local authority scores to the ranks of the current partition
-    for (auto &hMapAuthorityFlowWorldToLocalIterator: hMapAuthorityFlowWorldToLocal) {
-        long vertex = hMapAuthorityFlowWorldToLocalIterator.first;
-        float worldToLocalRank = hMapAuthorityFlowWorldToLocalIterator.second;
 
-        auto rankMapItr = rankMap.find(vertex);
-        if (rankMapItr != rankMap.end()) {
-            double existingRank = rankMapItr->second;
-            double finalRank = existingRank + worldToLocalRank;
-            rankMapItr->second = finalRank;
-        } else {
-            rankMap.insert(std::make_pair(vertex, worldToLocalRank));
+    map<long, long> inDegreeDistribution;
+
+    Utils utils;
+    std::string aggregatorFilePath = utils.getJasmineGraphProperty("org.jasminegraph.server.instance.datafolder");
+    std::string iddFilePath = aggregatorFilePath + "/" + graphID + +"_idd_" + partitionID;
+    ifstream dataFile;
+    dataFile.open(iddFilePath);
+
+    while(!dataFile.eof()) {
+        std::string str;
+        std::getline( dataFile, str);
+        std::stringstream buffer(str);
+        std::string temp;
+        std::vector<long> values;
+
+        while( getline( buffer, temp, '\t') ) {
+            values.push_back( ::strtod(temp.c_str(), 0));
+        }
+        if (values.size() == 2) {
+            long nodeID = values[0];
+            long iddValue = values[1];
+            inDegreeDistribution.insert(std::make_pair(nodeID, iddValue));
+        }
+    }
+
+    for (localGraphMapIterator = localGraphMap.begin();
+         localGraphMapIterator != localGraphMap.end(); ++localGraphMapIterator) {
+        auto inDegreeDistributionItr = inDegreeDistribution.find(localGraphMapIterator->first);
+
+        if (inDegreeDistributionItr != inDegreeDistribution.end()) {
+            long inDegree = inDegreeDistributionItr->second;
+            double authorityScore = (alpha * 1 + mu) * inDegree;
```

Comment:
Should `alpha` and `mu` be defined as constants?

Reply:
alpha is a actual variable here. mu is calculated based on alpha. Hence alpha is configured to be passed as a parameter.

Reply:
Ok